CMakeLists.txt

```diff
@@ -393,7 +393,11 @@ else()
 endif ()
 
 option (ENABLE_GWP_ASAN "Enable Gwp-Asan" ON)
-if (NOT OS_LINUX AND NOT OS_ANDROID)
+# We use mmap for allocations more heavily in debug builds, 
+# but GWP-ASan also wants to use mmap frequently, 
+# and due to a large number of memory mappings, 
+# it does not work together well.
+if ((NOT OS_LINUX AND NOT OS_ANDROID) OR (CMAKE_BUILD_TYPE_UC STREQUAL "DEBUG"))
```

Comment:
Actually this looks odd, since someone may expect from debug build to have as much checks as possible, instead,  maybe it is better to disable it in runtime? i.e.

```sh
GWP_ASAN_OPTIONS="Enabled=false"
```

@hanfei1991 what do you think?